lib/wallaroo/core/messages/channel_messages.pony

```diff
@@ -748,26 +770,132 @@ class val ForwardMsg[D: Any val] is ReplayableDeliveryMsg
     _msg_uid = msg_uid'
     _frac_ids = frac_ids'
 
-  fun target_id(): StepId => _target_id
   fun sender_name(): String => _sender_name
 
-  fun deliver(pipeline_time_spent: U64, target_step: Consumer,
-    producer_id: StepId, producer: Producer, seq_id: SeqId, route_id: RouteId,
-    latest_ts: U64, metrics_id: U16, worker_ingress_ts: U64): Bool
-  =>
+  fun val deliver(pipeline_time_spent: U64,
```

Comment:
Let's refactor this together where we have deliver for both `ForwardMsg` and `ForwardKeyedMsg` so that it no longer handles the routing logic but instead delegates to a new method on `DataRouter`, passing in the type arg we're carrying (`D`) and our data/metadata.

Reply:
@jtfmumm and I tried this and it ended up being harder than expected, so he opened issue #2257 to address it later.